tools/ci-operator-prowgen/main.go

```diff
@@ -0,0 +1,318 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	cioperatorapi "github.com/openshift/ci-operator/pkg/api"
+	kubeapi "k8s.io/api/core/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowkube "k8s.io/test-infra/prow/kube"
+)
+
+type options struct {
+	ciOperatorConfigPath string
+
+	fullRepoMode bool
+
+	ciOperatorConfigDir string
+	prowJobConfigDir    string
+
+	help    bool
+	verbose bool
+}
+
+func bindOptions(flag *flag.FlagSet) *options {
+	opt := &options{}
+
+	flag.StringVar(&opt.ciOperatorConfigPath, "source-config", "", "Path to ci-operator configuration file in openshift/release repository.")
+
+	flag.BoolVar(&opt.fullRepoMode, "full-repo", false, "If set to true, the generator will walk over all ci-operator config files in openshift/release repository and regenerate all component prow job config files")
+
+	flag.StringVar(&opt.ciOperatorConfigDir, "config-dir", "", "Path to a root of directory structure with ci-operator config files (ci-operator/config in openshift/release)")
+	flag.StringVar(&opt.prowJobConfigDir, "prow-jobs-dir", "", "Path to a root of directory structure with Prow job config files (ci-operator/jobs in openshift/release)")
+
+	flag.BoolVar(&opt.help, "h", false, "Show help for ci-operator-prowgen")
+	flag.BoolVar(&opt.verbose, "v", false, "Show verbose output")
+
+	return opt
+}
+
+// Generate a PodSpec that runs `ci-operator`, to be used in Presubmit/Postsubmit
+// Various pieces are derived from `org`, `repo`, `branch` and `target`.
+// `additionalArgs` are passed as additional arguments to `ci-operator`
+// Example
```

Comment:
I'd say let's not add these as the comments will rot fairly quick and we can achieve examples that can't rot through tests